lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -2629,6 +2629,181 @@ void mme_app_handle_nw_init_ded_bearer_actv_req(
   OAILOG_FUNC_OUT(LOG_MME_APP);
 }
 
+//------------------------------------------------------------------------------
+void _send_delete_dedicated_bearer_rsp(
+  struct ue_mm_context_s *ue_context_p,
+  bool delete_default_bearer,
+  ebi_t ebi[],
+  uint32_t num_bearer_context,
+  teid_t s_gw_teid_s11_s4,
+  gtpv2c_cause_value_t cause)
+{
+  itti_s11_nw_init_deactv_bearer_rsp_t *s11_deact_ded_bearer_rsp = NULL;
+  MessageDef *message_p = NULL;
+  uint32_t i = 0;
+
+  OAILOG_FUNC_IN(LOG_MME_APP);
+  if (ue_context_p == NULL) {
+    OAILOG_ERROR(LOG_MME_APP, " NULL UE context ptr\n");
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+  message_p =
+    itti_alloc_new_message(TASK_MME_APP,
+    S11_NW_INITIATED_DEACTIVATE_BEARER_RESP);
+  s11_deact_ded_bearer_rsp =
+    &message_p->ittiMsg.s11_nw_init_deactv_bearer_rsp;
+
+  if (message_p == NULL) {
+    OAILOG_ERROR(LOG_MME_APP,
+      "itti_alloc_new_message failed for"
+      "S11_NW_INITIATED_DEACTIVATE_BEARER_RESP\n");
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+  memset(s11_deact_ded_bearer_rsp, 0,
+    sizeof(itti_s11_nw_init_deactv_bearer_rsp_t));
+
+  s11_deact_ded_bearer_rsp->delete_default_bearer = delete_default_bearer;
+
+  if (delete_default_bearer) {
+    s11_deact_ded_bearer_rsp->lbi = calloc(1, sizeof(ebi_t));
+    *s11_deact_ded_bearer_rsp->lbi = ebi[0];
+    s11_deact_ded_bearer_rsp->bearer_contexts.bearer_contexts[0].
+      cause.cause_value = cause;
+    OAILOG_INFO(
+      LOG_MME_APP,
+      " Sending S11_PCRF_DED_BEARER_DEACTV_RESPONSE to SGW, (ebi = %d)\n",
+      ebi[0]);
+  } else {
+    for (i = 0; i < num_bearer_context; i++) {
+      s11_deact_ded_bearer_rsp->bearer_contexts.bearer_contexts[i].
+        eps_bearer_id = ebi[i];
+      s11_deact_ded_bearer_rsp->bearer_contexts.bearer_contexts[i].
+        cause.cause_value = cause;
+      OAILOG_INFO(
+        LOG_MME_APP,
+        " Sending S11_PCRF_DED_BEARER_DEACTV_RESPONSE to SGW, (ebi = %d)\n",
+        s11_deact_ded_bearer_rsp->bearer_contexts.bearer_contexts[i].
+          eps_bearer_id);
+    }
+  }
+  s11_deact_ded_bearer_rsp->bearer_contexts.num_bearer_context =
+    num_bearer_context;
+  s11_deact_ded_bearer_rsp->imsi = ue_context_p->imsi;
+  s11_deact_ded_bearer_rsp->s_gw_teid_s11_s4 = s_gw_teid_s11_s4;
+  itti_send_msg_to_task(TASK_SPGW, INSTANCE_DEFAULT, message_p);
+
+  OAILOG_FUNC_OUT(LOG_MME_APP);
+}
+
+
+/**
+ * This Function handles NW initiated
+ * Dedicated bearer Deactivation Request message from SGW
+ */
```

Comment:
Comments on functions belong in the .h file. Please do the same for all functions

Also, check capitalization for all added comments
"This function handles NW-initiated dedicated bearer deactivation request message from SGW"

Reply:
done